test/integration/clustercidr/ipam_test.go

```diff
@@ -138,6 +139,456 @@ func TestIPAMMultiCIDRRangeAllocatorType(t *testing.T) {
 	}
 }
 
+func TestIPAMMultiCIDRRangeAllocatorCIDRRelease(t *testing.T) {
+	// set the feature gate to enable MultiCIDRRangeAllocator
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.MultiCIDRRangeAllocator, true)()
+
+	_, kubeConfig, tearDownFn := framework.StartTestServer(t, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			// Disable ServiceAccount admission plugin as we don't have serviceaccount controller running.
+			opts.Admission.GenericAdmission.DisablePlugins = []string{"ServiceAccount", "TaintNodesByCondition"}
+			opts.APIEnablement.RuntimeConfig.Set("networking.k8s.io/v1alpha1=true")
+		},
+	})
+	defer tearDownFn()
+
+	clientSet := clientset.NewForConfigOrDie(kubeConfig)
+	sharedInformer := informers.NewSharedInformerFactory(clientSet, 1*time.Hour)
+
+	ipamController := booststrapMultiCIDRRangeAllocator(t, clientSet, sharedInformer)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go ipamController.Run(ctx.Done())
+	sharedInformer.Start(ctx.Done())
+
+	tests := []struct {
+		name             string
+		clusterCIDR      *networkingv1alpha1.ClusterCIDR
+		nodes            []*v1.Node
+		expectedPodCIDRs map[string][]string
+	}{
+		{
+			name:        "Pod CIDR release after node delete",
+			clusterCIDR: makeClusterCIDR("dualstack-cc", "192.168.0.0/23", "fd00:30:100::/119", 8, nodeSelector(map[string][]string{"ipv4": {"true"}, "ipv6": {"true"}})),
+			nodes: []*v1.Node{
+				makeNode("dualstack-node", map[string]string{"ipv4": "true", "ipv6": "true"}),
+				makeNode("dualstack-node-2", map[string]string{"ipv4": "true", "ipv6": "true"}),
+				makeNode("dualstack-node-3", map[string]string{"ipv4": "true", "ipv6": "true"}),
+			},
+			expectedPodCIDRs: map[string][]string{
+				"dualstack-node":   {"192.168.0.0/24", "fd00:30:100::/120"},
+				"dualstack-node-2": {"192.168.1.0/24", "fd00:30:100::100/120"},
+				"dualstack-node-3": {"192.168.0.0/24", "fd00:30:100::/120"},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.clusterCIDR != nil {
+				// Create the test ClusterCIDR
+				if _, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Create(context.TODO(), test.clusterCIDR, metav1.CreateOptions{}); err != nil {
+					t.Fatal(err)
+				}
+
+				// Wait for the ClusterCIDR to be created
```

Comment:
The ClusterCIDR object will be created on the Create operation, no need to check the object exists until you expect some external controller to modify some parameter and wait here for that